gateway-ha/src/test/java/io/trino/gateway/ha/TestGatewayHaMultipleBackend.java

```diff
@@ -158,6 +159,42 @@ public void testQueryDeliveryToMultipleRoutingGroups()
         assertThat(response4.body().string()).contains("http://localhost:" + routerPort);
     }
 
+    @Test
+    public void testTrinoClusterHostCookie()
+            throws Exception
+    {
+        RequestBody requestBody = RequestBody.create("SELECT 1", MediaType.get("application/json; charset=utf-8"));
+
+        // When X-Trino-Routing-Group is set in header, query should be routed to cluster under the routing group
+        Request requestWithoutCookie =
+                new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/v1/statement")
+                        .addHeader("X-Trino-User", "test")
+                        .post(requestBody)
+                        .addHeader("X-Trino-Routing-Group", "scheduled")
+                        .build();
+        Response responseOne = httpClient.newCall(requestWithoutCookie).execute();
+        assertThat(responseOne.body().string()).contains("http://localhost:" + routerPort);
+        List<Cookie> cookies = Cookie.parseAll(responseOne.request().url(), responseOne.headers());
+        Cookie cookie = cookies.stream().filter(c -> c.name().equals("trinoClusterHost")).collect(onlyElement());
+        assertThat(cookie.value()).isEqualTo("localhost");
+        // test with sending the request which includes trinoClusterHost in the cookie
+        // when X-Trino-Routing-Group is set in header, query should be routed to cluster under the routing group
+        Request requestWithCookie =
+                new Request.Builder()
+                        .url("http://localhost:" + routerPort + "/v1/statement")
+                        .addHeader("X-Trino-User", "test")
+                        .post(requestBody)
+                        .addHeader("X-Trino-Routing-Group", "scheduled")
+                        .addHeader("Cookie", "trinoClientHost=foo.example.com")
+                        .build();
+        Response responseTwo = httpClient.newCall(requestWithCookie).execute();
+        assertThat(responseTwo.body().string()).contains("http://localhost:" + routerPort);
+        cookies = Cookie.parseAll(responseTwo.request().url(), responseTwo.headers());
+        cookie = cookies.stream().filter(c -> c.name().equals("trinoClusterHost")).collect(onlyElement());
+        assertThat(cookie.value()).isEqualTo("localhost");
```

Comment:
> When X-Trino-Routing-Group is set in header, query should be routed to cluster under the routing group
> ...
> test with sending the request which includes trinoClusterHost in the cookie when X-Trino-Routing-Group is set in header, query should be routed to cluster under the routing group

Values in assertions between `requestWithoutCookie` and `requestWithCookie` are the same. I don't understand what you want to express in this comment.

Reply:
I'll fix the comment. The intent of this test is to make sure that the cookie is overriden correctly even though the request sends with some random value.